sound/soc/sdca/sdca_functions.c

```diff
@@ -1220,6 +1221,98 @@ static int find_sdca_entity_ge(struct device *dev,
+					struct fwnode_handle *entity_node,
+					struct sdca_entity *entity)
+{
+	struct sdca_entity_hide *hide = &entity->hide;
+	struct hid_descriptor hid_desc;
+	unsigned int delay, af_list[SDCA_MAX_FUNCTION_COUNT];
+	int nval, ret;
+	unsigned char *report_desc = NULL;
+
+	ret = fwnode_property_read_u32(entity_node,
+			"mipi-sdca-RxUMP-ownership-transition-maxdelay", &delay);
+	if (!ret)
+		hide->max_delay = delay;
+
+	nval = fwnode_property_count_u32(entity_node, "mipi-sdca-HIDTx-supported-report-ids");
+	if (nval > 0) {
+		hide->num_hidtx_ids = nval;
+		hide->hidtx_ids = devm_kcalloc(dev, hide->num_hidtx_ids,
+							sizeof(*hide->hidtx_ids), GFP_KERNEL);
+		if (!hide->hidtx_ids)
+			return -ENOMEM;
+
+		ret = fwnode_property_read_u32_array(entity_node,
+						"mipi-sdca-HIDTx-supported-report-ids",
+						hide->hidtx_ids,
+						hide->num_hidtx_ids);
+		if (ret < 0)
+			return ret;
+	} else {
+		hide->hidtx_ids = NULL;
+	}
+
+	nval = fwnode_property_count_u32(entity_node, "mipi-sdca-HIDRx-supported-report-ids");
+	if (nval > 0) {
+		hide->num_hidtx_ids = nval;
+		hide->hidtx_ids = devm_kcalloc(dev, hide->num_hidtx_ids,
+							sizeof(*hide->hidtx_ids), GFP_KERNEL);
+		if (!hide->hidtx_ids)
+			return -ENOMEM;
+
+		ret = fwnode_property_read_u32_array(entity_node,
+						"mipi-sdca-HIDRx-supported-report-ids",
+						hide->hidtx_ids,
+						hide->num_hidtx_ids);
+		if (ret < 0)
+			return ret;
+	} else {
+		hide->hidrx_ids = NULL;
+	}
+
+	nval = fwnode_property_count_u32(entity_node, "mipi-sdca-hide-related-audio-function-list");
+	if (nval <= 0) {
+		dev_err(dev, "%pfwP: HIDE - audio function numbers list missing: %d\n",
+			entity_node, nval);
+		return -EINVAL;
+	} else if (nval > SDCA_MAX_FUNCTION_COUNT) {
+		dev_err(dev, "%pfwP: maximum number of audio function exceeded\n", entity_node);
+		return -EINVAL;
+	}
+
+	fwnode_property_read_u32_array(entity_node, "mipi-sdca-hide-related-audio-function-list",
+		&af_list[0], nval);
+
+	nval = fwnode_property_count_u8(function_node, "mipi-sdca-hid-descriptor");
+	if (nval)
+		fwnode_property_read_u8_array(function_node, "mipi-sdca-hid-descriptor", (u8 *)&hid_desc, nval);
+	memcpy(&hide->hid_desc, &hid_desc, sizeof(hid_desc));
+
+	printk("HIDE: num_hid_desc = %d, bLength=0x%x \
+		bDescriptorType=0x%x, bcdHID=0x%x, bCountryCode=0x%x bNumDescriptors=0x%x \n",
+		nval, hide->hid_desc.bLength, hide->hid_desc.bDescriptorType,
+		hide->hid_desc.bcdHID, hide->hid_desc.bCountryCode, hide->hid_desc.bNumDescriptors);
+
+	if (hid_desc.bNumDescriptors) {
+		nval = fwnode_property_count_u8(function_node, "mipi-sdca-report-descriptor");
+		if (nval) {
+			report_desc = devm_kzalloc(dev, nval, GFP_KERNEL);
+			if (!report_desc)
+				return -ENOMEM;
+			hide->hid_report_desc = report_desc;
+			fwnode_property_read_u8_array(function_node, "mipi-sdca-report-descriptor", report_desc, nval);
+
+			/* add HID device */
+			sdca_add_hid_device(dev, entity_node, entity);
+		}
```

Comment:
Ditto here, pretty sure report-descriptor is required for HIDs so we should error out.

Reply:
The patch calls 'fwnode_property_count_u8' to check whether the report descriptor exists.
So the code will add the HID device when the HID/Report descriptors exist.

Reply:
I am with @charleskeepax, we need to make sure all reads are tested. Checking the size of the array is not good enough. we had a similar discussion for mipi properties with @andy-shev a long time ago. Don't assume all reads succeed.

Reply:
For clarity, I think the check on the call to count should be enough to ensure the read itself succeeds, unless I have missed something. My comment was really more that if the widget is a HIDE one (which is is if we are here in the code) then "mipi-sdca-report-descriptor" isn't optional, so we should return an error if it doesn't exist.